src/main/java/io/jenkins/plugins/jfrog/configuration/JenkinsProxyConfiguration.java

```diff
@@ -0,0 +1,63 @@
+package io.jenkins.plugins.jfrog.configuration;
+
+import hudson.util.Secret;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang3.StringUtils;
+import org.jfrog.build.client.ProxyConfiguration;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents the proxy configuration that is retrieved from the Jenkins settings.
+ * This information is populated from the Plugin Manager's Advanced section.
+ * Because of the Jenkins.get() used in the constructor, you can only create it on the master, not on an agent.
+ */
+public class JenkinsProxyConfiguration extends ProxyConfiguration {
+    private static final Pattern HOST_NAME_PATTERN = Pattern.compile("^.*?://?([\\w.]+).*");
+    public List<Pattern> noProxyHostPatterns;
+    public String noProxy;
+
+    public JenkinsProxyConfiguration() {
+        hudson.ProxyConfiguration proxy = Jenkins.get().getProxy();
+        if (proxy == null) {
+            return;
+        }
+        this.host = proxy.getName();
+        this.port = proxy.getPort();
+        this.username = proxy.getUserName();
+        this.password = Secret.toString(proxy.getSecretPassword());
+        this.noProxy = proxy.getNoProxyHost();
+        this.noProxyHostPatterns = proxy.getNoProxyHostPatterns();
+    }
+
+    /**
+     * Return true if the proxy is configured.
+     *
+     * @return true if the proxy is configured.
+     */
+    public boolean isProxyConfigured() {
+        return StringUtils.isNotBlank(host);
+    }
+
+    /**
+     * Return true if the host is matched to one of the 'No Proxy Host' patterns.
```

Comment:
Return true if the host is matched to one of the 'No Proxy Host' patterns.
-->
Return true if the host matches one of the 'No Proxy Host' patterns.